homeassistant/components/modbus/climate.py

```diff
@@ -204,11 +220,41 @@ def __init__(
                     self._attr_fan_modes.append(fan_mode)
 
         else:
-            # No HVAC modes defined
+            # No FAN modes defined
             self._fan_mode_register = None
             self._attr_fan_mode = FAN_AUTO
             self._attr_fan_modes = [FAN_AUTO]
 
+        if CONF_SWING_MODE_REGISTER in config:
+            self._attr_supported_features = (
+                self._attr_supported_features | ClimateEntityFeature.SWING_MODE
+            )
+            mode_config = config[CONF_SWING_MODE_REGISTER]
+            self._swing_mode_register = mode_config[CONF_ADDRESS]
+            self._attr_swing_modes = cast(list[str], [])
+            self._attr_swing_mode = None
+            self._swing_mode_mapping_to_modbus: dict[str, int] = {}
+            self._swing_mode_mapping_from_modbus: dict[int, str] = {}
+            mode_value_config = mode_config[CONF_SWING_MODE_VALUES]
+            for swing_mode_kw, swing_mode in (
+                (CONF_SWING_MODE_SWING_ON, SWING_ON),
+                (CONF_SWING_MODE_SWING_OFF, SWING_OFF),
+                (CONF_SWING_MODE_SWING_HORIZ, SWING_HORIZONTAL),
+                (CONF_SWING_MODE_SWING_VERT, SWING_VERTICAL),
+                (CONF_SWING_MODE_SWING_BOTH, SWING_BOTH),
+            ):
+                if swing_mode_kw in mode_value_config:
+                    value = mode_value_config[swing_mode_kw]
+                    self._swing_mode_mapping_from_modbus[value] = swing_mode
+                    self._swing_mode_mapping_to_modbus[swing_mode] = value
+                    self._attr_swing_modes.append(swing_mode)
+
+        else:
+            # No SWING modes defined
+            self._swing_mode_register = None
```

Comment:
Remove the "else" and define it before "if" that simplify the "if" section as well, and is less confusing.

Reply:
Apart from that I am not sure that SWING_OFF is a correct default ! because it will be missing in the the 2 mapping arrays ??

Reply:
I think that it is not required to initialize the `self._attr_swing_modes ` and ` self._attr_swing_mode = None` in that phase, because at that time, swing support in not available, so the only thing to initialize is the register.